### PR TITLE
Update Israel holidays

### PR DESCRIPTION
**Pull request by arkid15r**
## Proposed change

Resolves https://github.com/vacanza/python-holidays/issues/1549

Update Israel holidays:
  - Add Hebrew calendar holiday name constatns
  - Add l10n support: en_US, uk
  - Update README, snapshots, tests

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've added references to all holidays information sources used in this PR
- [x] The code style looks good: `make pre-commit` command generates no changes
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/locale/uk/LC_MESSAGES/IL.po [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+msgid "יום כיפור"
+msgstr "Йом Кіпур"
+
+#. Sukkot (Feast of Tabernacles).
+msgid "סוכות"
+msgstr "Суккот"
+
+#. Simchat Torah / Shemini Atzeret.
+msgid "שמחת תורה/שמיני עצרת"
+msgstr "Симхат Тора / Восьмий день зборів"
```
Comment (KJhellico) [inline holidays/locale/uk/LC_MESSAGES/IL.po:32]:
```suggestion
msgstr "Сімхат Тора / Восьмий день зборів"
```
In [Wiki](https://uk.wikipedia.org/wiki/%D0%A1%D1%96%D0%BC%D1%85%D0%B0%D1%82_%D0%A2%D0%BE%D1%80%D0%B0) and most online references in Ukrainian.

File: holidays/locale/uk/LC_MESSAGES/IL.po [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+msgid "יום הזיכרון לחללי מערכות ישראל ונפגעי פעולות האיבה"
+msgstr "День памʼяті"
+
+#. Yom Yerushalayim (Jerusalem Day).
+msgid "יום ירושלים"
+msgstr "День Єрусалиму"
+
+#. Tisha B'Av (Tisha B'Av, fast).
+msgid "תשעה באב"
+msgstr "Тиша Бе Ав"
```
Comment (KJhellico) [inline holidays/locale/uk/LC_MESSAGES/IL.po:76]:
```suggestion
msgstr "Тиша Бе-Ав"
```

File: holidays/locale/uk/LC_MESSAGES/IL.po [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+#. Hanukkah.
+msgid "חנוכה"
+msgstr "Ханука"
+
+#. Ta`anit Ester (Fast of Esther).
+msgid "תענית אסתר"
+msgstr "Таніт Естер"
```
Comment (KJhellico) [inline holidays/locale/uk/LC_MESSAGES/IL.po:84]:
```suggestion
msgstr "Піст Естер"
```
Or maybe `"Тааніт-Естер"`?

**Review by KJhellico — COMMENTED** [review]
I offer a few corrections to Ukrainian translation, but don't insist on them. :)

**Comment by arkid15r** [comment]
> I offer a few corrections to Ukrainian translation, but don't insist on them. :)

I've gladly accepted your suggestions 👍

**Comment by arkid15r** [comment]
Hi @sjspielman, could you review [IL.po](https://github.com/vacanza/python-holidays/pull/1557/files#diff-eae153f85683f1f1fd93c1a9f223c30a038891a0b8bd7cc75520035b5b8749d5) file? 

There is also `yi` localization but I'm open for removing it depending on your recommendations.

Thank you!

File: holidays/countries/israel.py [hunk 4]
```diff
@@ -27,15 +39,18 @@
 
 class Israel(ObservedHolidayBase):
     country = "IL"
-    observed_label = "%s (Observed)"
+    default_language = "he"
+    # %s (Observed).
+    observed_label = tr("(נצפה) %s")
```
Comment (sjspielman) [inline holidays/countries/israel.py:44]: I imagine you got this from the "observed" translation? I'm not sure if this is the right word honestly..

File: holidays/countries/israel.py [hunk 5]
```diff
@@ -83,42 +101,42 @@ def _populate_optional_holidays(self):
...
-            if not is_obs:
-                self._add_holiday(name, memorial_day_dt)
+            is_observed, _ = self._add_observed(remembrance_day_dt, name, rule)
+            if not is_observed:
+                self._add_holiday(name, remembrance_day_dt)
 
-        # Jerusalem Day
-        self._add_holiday("Jerusalem Day", self._get_holiday("LAG_BAOMER") + td(days=+10))
+        # Yom Yerushalayim (Jerusalem Day).
+        self._add_holiday(tr("יום ירושלים"), self._get_holiday(LAG_BAOMER) + td(days=+10))
```
Comment (sjspielman) [inline holidays/countries/israel.py:133]: this would not have been celebrated before 1967

File: holidays/countries/israel.py [hunk 6]
```diff
@@ -83,42 +101,42 @@ def _populate_optional_holidays(self):
...
-            self._add_holiday("Passover", passover_dt + td(days=offset))
+            # Chol HaMoed Pesach (Passover holiday).
+            self._add_holiday(tr("חול המועד פסח"), passover_dt + td(days=offset))
 
-        # Memorial Day
         if self._year >= 1963:
-            name = "Memorial Day"
-            memorial_day_dt = self._get_holiday("INDEPENDENCE_DAY") + td(days=-1)
+            # Yom Hazikaron (Fallen Soldiers and Victims of Terrorism Remembrance Day).
+            name = tr("יום הזיכרון לחללי מערכות ישראל ונפגעי פעולות האיבה")
```
Comment (sjspielman) [inline holidays/countries/israel.py:123]: Generally this is just יום הזיכרון
but the rest is fine

File: holidays/countries/israel.py [hunk 7]
```diff
@@ -128,35 +146,35 @@ def _populate_school_holidays(self):
...
+        # Purim.
+        self._add_holiday(tr("פורים"), purim_dt)
 
-        # Passover (Chol HaMoed Pesach)
-        passover_dt = self._get_holiday("PASSOVER")
+        passover_dt = self._get_holiday(PASSOVER)
         for offset in range(1, 6):
-            self._add_holiday("Passover", passover_dt + td(days=offset))
+            # Chol HaMoed Pesach (Passover holiday).
+            self._add_holiday(tr("חול המועד פסח"), passover_dt + td(days=offset))
```
Comment (sjspielman) [inline holidays/countries/israel.py:174]: for organization, move this up to be next to `passover_dt`.

File: holidays/locale/en_US/LC_MESSAGES/IL.po [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+msgid "יום העצמאות"
+msgstr "Independence Day"
+
+#. Shavuot.
+msgid "שבועות"
+msgstr "Shavuot"
+
+#. Chol HaMoed Sukkot (Feast of Tabernacles holiday).
+msgid "חול המועד סוכות"
+msgstr "Sukkot holiday"
```
Comment (sjspielman) [inline holidays/locale/en_US/LC_MESSAGES/IL.po:52]: I'd move this up around line 26. It's common that Day 1 of a holiday is somehow distinguished from the rest of the days, if it's >1 day. So, Sukkot is day 1, and the remaining days of the holiday are this one.

File: holidays/locale/en_US/LC_MESSAGES/IL.po [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+msgid "סיגד"
+msgstr "Sigd"
+
+#. Purim.
+msgid "פורים"
+msgstr "Purim"
+
+#. Chol HaMoed Pesach (Passover holiday).
+msgid "חול המועד פסח"
+msgstr "Pesach holiday"
```
Comment (sjspielman) [inline holidays/locale/en_US/LC_MESSAGES/IL.po:64]: again move up to be with the other Pesach holidays for clearer organization

File: holidays/locale/yi/LC_MESSAGES/IL.po
Comment (sjspielman) [inline holidays/locale/yi/LC_MESSAGES/IL.po:1]: no need for this file

Comment (sjspielman) [reply holidays/locale/yi/LC_MESSAGES/IL.po:1]: (people who speak yiddish as a native language almost always speak hebrew, too!)

File: holidays/countries/israel.py (hunk 6 above)
Comment (sjspielman) [reply holidays/countries/israel.py:123]: but that might be just how i refer to it! you are probably right that this is the official name!

File: holidays/countries/israel.py (hunk 4 above)
Comment (arkid15r) [reply holidays/countries/israel.py:52]: That's right. Is current version translation somewhat irrelevant? I'd accept your best shot for this if any.

File: holidays/countries/israel.py (hunk 5 above)
Comment (arkid15r) [reply holidays/countries/israel.py]: Done

File: holidays/countries/israel.py (hunk 7 above)
Comment (arkid15r) [reply holidays/countries/israel.py:183]: That's a different category (public vs school holidays). I'd definitely have them combined if they were same kind.

File: holidays/locale/en_US/LC_MESSAGES/IL.po (hunk 8 above)
Comment (arkid15r) [reply holidays/locale/en_US/LC_MESSAGES/IL.po]: Good point. I hope `.po` files auto-generation process will not mess this up later.

File: holidays/locale/en_US/LC_MESSAGES/IL.po (hunk 9 above)
Comment (arkid15r) [reply holidays/locale/en_US/LC_MESSAGES/IL.po]: Done

File: holidays/locale/yi/LC_MESSAGES/IL.po
Comment (arkid15r) [reply holidays/locale/yi/LC_MESSAGES/IL.po:1]: ✔️

**Review by arkid15r — COMMENTED** [review]
Thanks a lot for reviewing this!

**Review by KJhellico — APPROVED** [review]
LGTM 🇮🇱

**Comment by arkid15r** [comment]
> LGTM 🇮🇱

Yay! You've got the flags!

**Pull request merged**